lms/djangoapps/certificates/views/webview.py

```diff
@@ -336,7 +343,7 @@ def _get_user_certificate(request, user, course_key, course, preview_mode=None):
             user_certificate = GeneratedCertificate(
                 mode=preview_mode,
                 verify_uuid=unicode(uuid4().hex),
-                modified_date=datetime.now().date()
+                modified_date=course.certificate_available_date if not course.self_paced else datetime.now().date()
```

Comment:
Is it possible that the course won't have a `certificate_available_date` at this point?